htmresearch/encoders/language_encoder.py

```diff
@@ -183,11 +183,14 @@ def sparseUnion(self, counts):
 
     @param counts     (Counter)   A count of the ON bits for the union bitmap.
 
-    @return           (list)      A sparsified union bitmap.
+    @return           (tuple)      A sparsified union bitmap.
     """
+    positionsSortedByCount = tuple(x[1] for x in counts.getNonZerosSorted())
```

Comment:
https://github.com/numenta/nupic.core/blob/008a4ba51dd6e1303b9b125a9602a91734db0b8d/src/nupic/math/SparseMatrix.hpp#L4304

According to the docs for `std::partial_sort()`:

> The order of equal elements is not guaranteed to be preserved. The order of the remaining elements in the range [middle, last) is unspecified. The first version uses operator< to compare the elements, the second version uses the given comparison function comp.

Maybe we should consider revising the implementation to use `std::stable_sort()`?